client/src/modules/reports/generate/inventory_file/inventory_file.config.js

```diff
@@ -0,0 +1,87 @@
+angular.module('bhima.controllers')
+  .controller('inventory_fileController', InventoryFileConfigController);
+
+InventoryFileConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state',
+  'LanguageService', 'InventoryService', '$timeout', 'moment',
+];
+
+function InventoryFileConfigController($sce, Notify, SavedReports, AppCache, reportData, $state,
+  Languages, Inventory, $timeout, moment) {
+  var vm = this;
+  var cache = new AppCache('configure_inventory_file_report');
+  var reportUrl = 'reports/stock/inventory';
+
+  vm.previewGenerated = false;
+
+  vm.dateTo = new Date();
+
+  // chech cached configuration
+  checkCachedConfiguration();
+
+  Inventory.read()
+  .then(function (rows) {
+    vm.inventories = rows;
+  })
+  .catch(Notify.handleError);
+
+  vm.setupDepot = function setupDepot(depot) {
+    vm.depot = depot;
+  };
+
+  vm.clearPreview = function clearPreview() {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.preview = function preview(form) {
+    var options;
+    var params;
+
+    if (form.$invalid || !vm.depot || !vm.inventory) { return 0; }
```

Comment:
Can you not make `vm.depot` or `vm.inventory` required, and then just use `form.$invalid`?